mfsetup/grid.py

```diff
@@ -728,9 +728,18 @@ def setup_structured_grid(xoff=None, yoff=None, xul=None, yul=None,
     if parent_model is not None and not snap_to_NHG:
 
         # get location of coinciding cell in parent model for upper left
-        pi, pj = parent_model.modelgrid.intersect(xul, yul)
-        verts = np.array(parent_model.modelgrid.get_cell_vertices(pi, pj))
-        xul, yul = verts[:, 0].min(), verts[:, 1].max()
+        # navigating the vertices of the parent cell depends on rotation angle
+        if rotation == 0:
+            pi, pj = parent_model.modelgrid.intersect(xul, yul)
+            verts = np.array(parent_model.modelgrid.get_cell_vertices(pi, pj))
+            xul, yul = verts[:, 0].min(), verts[:, 1].max()
+        else:
+            # if there is rotation, need to make sure we locate within the actual upper left.
+            pi, pj = parent_model.modelgrid.intersect(xul + (delc_grid/100), yul)
```

Comment:
what about using the cell center here (``xul + 0.5 * delr_grid, yul + 0.5 * delc_grid``)? That the corner will end up aligning with whatever parent cell the inset cell center is within. Also I think we want to use delr for x (spacing *along* a row) and delc for y (spacing *along* a column).

Reply:
ugh - always  getting burned by the `delr` `delc` meaning. but....your suggestion can end up not in the center and going too high in y-space, resulting in snapping to the wrong cell. I think we need only a tiny nudge along only one axis (x-axis works for sure for CCW rotation).